.travis.yml

```diff
@@ -1,14 +1,73 @@
 sudo: false
 language: rust
 rust:
-    - 1.29.1
-branches:
-    only:
-        - master
-notifications:
-    email:
-        on_success: change
-        on_failure: always
+- 1.29.1
+
+stages:
+  - compile
+  - test
+  - npm publish
+
+jobs:
+  include:
+    # commits with [build] in them get built on a single arch
+  - name: "compile and test"
+    stage: "compile"
+    os: linux
+    env: TRAVIS_NODE_VERSION="8"
+    if: commit_message =~ /[build]/
+    script:
+      - yarn install --ignore-scripts
+      - yarn run compile
+      - yarn test
+  
+  # PRs and commits to master build on all target arches
+  # if this also has a release tag, the resultant binary will be uploaded to github
+  - name: "Linux - Node 8"
+    os: linux
+    env: TRAVIS_NODE_VERSION="8"
+    if: branch = master OR type = pull_request
+  - name: "OSX - Node 8"
+    os: osx
+    env: TRAVIS_NODE_VERSION="8"
+    if: branch = master OR type = pull_request
+  - name: "Linux - Node 10"
+    os: linux
+    env: TRAVIS_NODE_VERSION="10"
+    if: branch = master OR type = pull_request
+  - name: "OSX - Node 10"
+    os: osx
+    env: TRAVIS_NODE_VERSION="10"
+    if: branch = master OR type = pull_request
+
+  # Publish to npm only on release tag
+  - stage: "npm publish"
+    os: linux
+    env: 
+      - TRAVIS_NODE_VERSION="8"
+      - secure: 
+          DSyA0KDZogwoaekLIPmR9wll1Unpyftn6l0nRDgkPv8MmTTu2a84JO17EP0zaXtBQlfTqp0x33bDHIahtr1ItAi931TYW5279B16EZ1gM3GG0pCHFN6E7QznBYrAEfrtnUKaC1E85ivVcPlF7rtDy7nEnzlnE0Lc5dPk57LSGsQHxJBPaVJ2SCVS4yWArSzkC1XyUeHq37ApDrdfOkObbV/lSKn1wj/b/oIZW5aI4C+24X+rfjOa9CvwF6fst1rOEB/XLg/9OIyAb8b9KjsjMReX6i39KyruWODzIfoOjIXLDXQ2Z8HnBJIBDg4P36CqXIZcudecVdTG0S7oLkhBNnH67W+MirI4QHXpsUGL4jiLG8KfzZADpOE1aI44SY+XODrMYDqNbkNz1iD+fVuWhplywGIa3KHRlJxqfcT5T3CSOcsX1VY0gnCIK+5pyl71MM7Uacg++ZuICoLdMXULPypMViBLpCB4UBUvGsjBbiTXmaX4MWMgkb+LN1WKVwFAoNa1T6v7SoBsRQmeZ/yYlRR2p8wCBU7BjF7no4xX++725A8mOELF/To27Opc2v7zgXRKQUTEclZ6a/xCu3u8kryNsRRQv9Iv+LcLaFBMz39/qMg+DmKvriITY8RUBzjbNddget4m3caGbq6Hzu0RKvoXzC8HH83E49pVkFiH0R0=
+    if: branch = master AND tag = /^v\d+\.\d+\.\d+.*$/
+    before_install:
+      - npm config set //registry.npmjs.org/:_authToken=$NPM_TOKEN
+    script: 
+      - echo "Deploying to NPM..."
+      - node publish.js --publish
+    deploy: echo "Skipping upload of binary artifacts"
+
+
+install:
+  # install our own yarn to make things work on osx
+  - curl -o- -L https://yarnpkg.com/install.sh | bash -s -- --version 1.10.1
+  - export PATH=$HOME/.yarn/bin:$PATH
+  
+  # install our own nodejs to get a reasonable version
+  - rm -rf ~/.nvm && git clone https://github.com/creationix/nvm.git ~/.nvm && (cd ~/.nvm && git checkout `git describe --abbrev=0 --tags`) && source ~/.nvm/nvm.sh && nvm install $TRAVIS_NODE_VERSION
+
+# notifications:
+#   email:
+    # on_success: change
+#    on_failure: always
```

Comment:
Will these be uncommented once we're ready to merge this?

Reply:
Yep. It's on the PR's TODO list.